commands/import.go

```diff
@@ -184,46 +208,105 @@ func (i *Importer) run() error {
 		return err
 	}
 
+	i.res = resources.InitResources(i.configFile)
+
 	// set up target database
 	i.res.DB.SelectDB(i.targetDatabase)
 
-	// get settings from an existing database
+	// set up the rolling configuration
+	// grab the current rolling settings from the MetaDB
 	exists, isRolling, currChunk, totalChunks, err := i.res.MetaDB.GetRollingSettings(i.targetDatabase)
 	if err != nil {
 		return cli.NewExitError(fmt.Errorf("\n\t[!] Error while reading existing database settings: %v", err.Error()), -1)
 	}
-	// determine the new rolling settings based on current and supplied arguments
-	rollingCfg, err := setRolling(exists, isRolling, currChunk, totalChunks,
-		i.rolling, i.currentChunk, i.totalChunks, i.res.Config.S.Rolling.DefaultChunks)
+
+	// validate the user given flags against the rolling settings from the MetaDB
+	// and determine the rolling configuration
+	rollingCfg, err := parseFlags(
+		exists, isRolling, currChunk, totalChunks,
+		i.userRolling, i.userCurrChunk, i.userTotalChunks, i.res.Config.S.Rolling.DefaultChunks,
+		i.deleteOldData,
+	)
 	if err != nil {
 		return cli.NewExitError(err.Error(), -1)
 	}
-	// preserve the default chunks setting (even though we don't use it after this currently)
-	rollingCfg.DefaultChunks = i.res.Config.S.Rolling.DefaultChunks
 	i.res.Config.S.Rolling = rollingCfg
 
 	importer := parser.NewFSImporter(i.res, i.threads, i.threads, i.importFiles)
 	if len(importer.GetInternalSubnets()) == 0 {
 		return cli.NewExitError("Internal subnets are not defined. Please set the InternalSubnets section of the config file.", -1)
 	}
 
+	indexedFiles := importer.CollectFileDetails()
+	// if no compatible files for import were found, exit
+	if len(indexedFiles) == 0 {
+		return cli.NewExitError("No compatible log files found", -1)
+	}
```

Comment:
Added check here to make sure some compatible files are actually found. This fixes 2 and 3 (along with a piece elsewhere).
```
$ ./rita import --delete ignore/empty rtest

No compatible log files found

$ ./rita import --delete ignore/unsupported/weird.log rtest

No compatible log files found

$ ./rita import --delete ignore/unsupported rtest

No compatible log files found
```